Realm/Realm/Realm.cs

```diff
@@ -724,6 +734,30 @@ public bool Refresh()
             return SharedRealmHandle.Refresh();
         }
 
+        /// <summary>
+        /// Waits for the realm and outstanding objects to be updated to a newer version.
+        /// </summary>
+        /// <returns>An awaitable <see cref="Task"/>.</returns>
+        private Task WaitForChangesAsync()
+        {
+            if (SynchronizationContext.Current == null)
+            {
+                Refresh();
+                return Task.CompletedTask;
+            }
+
+            var tcs = new TaskCompletionSource<object>();
+            RealmChangedEventHandler handler = null;
+            handler = (sender, e) =>
+            {
+                ((Realm)sender).RealmChanged -= handler;
```

Comment:
I'm a bit hazy on the subtleties of doing it like this (via the `sender`) rather than unsubscribing `this.RealmChanged`. Is it to avoid capturing `this` in the closure and what would the harm be if we did capture it?

Reply:
Exactly - I didn’t want to capture `this` and create a reference cycle. If we capture it and the wvent never fires then the realm will leak.